.github/workflows/test.yml

```diff
@@ -0,0 +1,28 @@
+name: 'Test'
+on:
+  pull_request:
+    types: [ opened, reopened, synchronize ]
+
+jobs:
+  test:
+    name: 'Unit and Integration Tests'
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+
+      - name: Use Java 17
+        uses: actions/setup-java@v3
+        with:
+          distribution: 'temurin'
+          java-version: '17'
```

Comment:
I do wonder if adding the `cache: 'gradle'` is worth it here.

Reply:
we can do 
https://github.com/marketplace/actions/gradle-cache
https://github.com/marketplace/actions/docker-layer-caching